tests/integration/rdb.tcl

```diff
@@ -416,4 +416,258 @@ start_server {} {
     } {OK}
 }
 
+set server_path [tmpdir "server.partial-hfield-exp-test"]
+
+# verifies writing and reading hash key with expiring and persistent fields
+start_server [list overrides [list "dir" $server_path]] {
+    foreach type {listpack dict} {
+        test "hash field expiration save and load rdb one expired field, ($type)" {
+            if {$type eq "dict"} {
+                r config set hash-max-listpack-entries 0
+            } else {
+                r config set hash-max-listpack-entries 512
+            }
+
+            r FLUSHALL
+
+            r HMSET key a 1 b 2 c 3 d 4
+            r HEXPIREAT key 2524600800 2 a b
+            r HPEXPIRE key 100 1 d
```

Comment:
```suggestion
```suggestion
    foreach {type lp_max_entries} {listpackex 512 hashtable 0} {
        test "hash field expiration save and load rdb one expired field, ($type)" {
            r config set hash-max-listpack-entries $lp_max_entries
            r FLUSHALL

            r HMSET key a 1 b 2 c 3 d 4
            r HEXPIREAT key 2524600800 2 a b
            r HPEXPIRE key 100 1 d
            assert_encoding $type key 
```